react/src/components/UsageHistoryStatistics.tsx

```diff
@@ -0,0 +1,196 @@
+import { useThemeMode } from '../hooks/useThemeMode';
+import useUserStats, {
+  Period,
+  UnitHint,
+  UsageHistoryKey,
+} from '../hooks/useUserStats';
+import Flex from './Flex';
+import { Column, ColumnConfig, Line, LineConfig } from '@ant-design/charts';
+import { Card, theme } from 'antd';
+import { format } from 'date-fns';
+
+type UsageHistoryTitle =
+  | 'Sessions'
+  | 'CPU'
+  | 'Memory'
+  | 'GPU'
+  | 'IO-Read'
+  | 'IO-Write';
+interface GraphContainerProps {
+  title: UsageHistoryTitle;
+  graph: 'line' | 'column';
+  config: LineConfig | ColumnConfig;
+  height?: number;
+}
+const GraphContainer = ({
+  title,
+  graph,
+  config,
+  height = 200,
+}: GraphContainerProps) => {
+  return (
+    <Card type="inner" title={title} style={{ width: '100%' }}>
+      {graph === 'line' ? (
+        <Line height={height} {...(config as LineConfig)} />
+      ) : (
+        <Column height={height} {...(config as ColumnConfig)} />
+      )}
+    </Card>
+  );
+};
+
+const lineConfig = (
+  data: any,
+  period: Period,
+  unitHint: UnitHint,
+  isDarkMode: boolean,
+): LineConfig => ({
+  data,
+  xField: 'date',
+  yField: 'value',
+  point: {},
+  interaction: {
+    tooltip: {
+      marker: false,
+    },
+  },
+  style: {
+    lineWidth: 2,
+  },
+  axis: {
+    x: {
+      labelAutoHide: true,
+      tickFilter: (_: any, index: any) =>
+        index % (period === '1D' ? 12 : 48) === 0,
+    },
+    y: {
+      title: unitHint,
+    },
+  },
+  theme: isDarkMode ? 'dark' : 'light',
+});
+
+const columnConfig = (
+  data: any,
+  period: Period,
+  unitHint: UnitHint,
+  isDarkMode: boolean,
+): ColumnConfig => ({
+  data,
+  xField: 'date',
+  yField: 'value',
+  axis: {
+    x: {
+      labelAutoHide: true,
+      tickFilter: (_: any, index: any) =>
+        index % (period === '1D' ? 12 : 48) === 0,
+    },
+    y: {
+      title: unitHint,
+    },
+  },
+  animate: { enter: { type: 'growInY' } },
+  theme: isDarkMode ? 'dark' : 'light',
+});
+
+const byteConverter = {
+  toB: (bytes: number) => bytes,
+  toKB: (bytes: number) => bytes / 1024,
+  toMB: (bytes: number) => bytes / (1024 * 1024),
+  toGB: (bytes: number) => bytes / (1024 * 1024 * 1024),
+  toTB: (bytes: number) => bytes / (1024 * 1024 * 1024 * 1024),
+  log1024: (n: number) => (n <= 0 ? 0 : Math.log(n) / Math.log(1024)),
+  readableUnit: function (bytes: number) {
+    return ['B', 'KB', 'MB', 'GB', 'TB'][Math.floor(this.log1024(bytes))];
+  },
+};
```

Comment:
Please use GiB (binary) for memory. There are several functions to help with unit conversion. ``convertBinarySizeUnit`, `convertDecimalSizeUnit` 
You can see how to use it in the test code.
https://github.com/lablup/backend.ai-webui/blob/3687e5b50f8d897d7254e94b164bd47dc2d7a79f/react/src/helper/index.test.tsx